### DISCUSSION TITLE
Added CPU temp, mem load and other Orange Pi OS specific info to Diagnotics web page (diagnose.sh)

### DISCUSSION THREAD

**Discussion by irsx02** [marcone/teslausb#821] [Show and tell]
On the Diagnostic web page I wanted to see my CPU temp for my Orange Pi Zero 2 W.   I notice that page was just capturing text log directly from "_setup-teslausb diagnose_."   I also know that the Orange Pi OS (Bookworm) has a nice screen that shows the CPU temp, mem usage, uptime and other useful stats.   

 Lets combine both information sets so we see the CPU temp, mem usage, and other diagnostics info.    Here's what I did.

First, I changed the root partition to writable` /root/bin/remountfs_rw` 
Next, I had to download ansi2txt  via `apt-get colorized-logs`    The ansi2txt program is needed to strip the ansi colors from the OrangePi system info.
Then I went over to `/var/www/html/cgi-bin` and `nano diagnose.sh`  
I modified the file to look like this;

```
#!/bin/bash

(cat /var/run/motd.dynamic | ansi2txt)  &> /tmp/diagnostics.txt
(sudo /root/bin/setup-teslausb diagnose) >> /tmp/diagnostics.txt

"$(dirname "$0")/reload.sh" "Sync triggered"

```
 
Basically, two changes; I added the _(cat /var/run._.. to run the Orange Pi message of the day (aka CPU temp, up time, mem usage)
And my other change was on the next line switching a & > over to >> so that the diagnostic.txt gets appended and not overwritten.

So now I get the Orange Pi motd info into the diagnostic log.  Because  I want to see how the CPU temp does in the summer inside a glovebox.     

Heres' a sample of what is on my diagnostic web page:
```
 ___  ____ ___   _____             ______        __
 / _ \|  _ \_ _| |__  /___ _ __ ___|___ \ \      / /
| | | | |_) | |    / // _ \ '__/ _ \ __) \ \ /\ / / 
| |_| |  __/| |   / /|  __/ | | (_) / __/ \ V  V /  
 \___/|_|  |___| /____\___|_|  \___/_____| \_/\_/   
                                                    
Welcome to Orange Pi 1.0.0 Bookworm with Linux 6.1.31-sun50iw9

System load:   74%           	Up time:       0 min	
Memory usage:  24% of 981M   	IP:	       192.168.0.123
CPU temp:      48°C           	Usage of /:    61% of 2.9G   	

====== summary ======
hardware: OrangePi Zero2 W
OS: Debian GNU/Linux 12 (bookworm)
(and normal info from the diagnotic page...) 
```

I hope this info helped another Orange Pi Zero 2 W (or other OrangePI OS) users out there.

**Comment by irsx02** [comment]
**IMPORTANT UPDATE**   It looks like that `cat /var/run/motd.dynamic` will not give you live-up-to-date CPU temperature.  It saves only temp at boot up, and does not update afterwards.  It is not the CPU temp I'm looking for.

**Comment by marcone** [comment]
Does the "Tools" tab in the web interface not show the temperature for you?

**Comment by irsx02** [comment]
@marcone   well, I must have missed that Tools tab - yes, the Tools tabdoes show the CPU temperature.

I meant, I wanted to see live updates on the memory usage ;)
